### PR TITLE
Implement map text

### LINKED ISSUE #28677: Maptext with richtext support for the tutorial map
## Description
Allows the use of tags like `KeyBindTag` for tutorial "decals"
This is used for tutorial text that looks like decals on the tutorial map.
### Example
tutorial_text_example

### PR DESCRIPTION
**Pull request by juliangiebel**
## About the PR

This PR adds the `MapText` component and a corresponding marker entity which allows you to arbitrarily display text in world.
The component lets you change the font, size, color and offset so you can add it to anything as a sort of in world label.

Admins can just spawn a map text marker or add the component the any entity to display funny text.
## Why / Balance

This is primarily for the tutorial terminal.
## Technical details

Map text uses an overlay to render the text on the screen drawing layers. 
Text outside of the viewpoint isn't drawn.
## Media
Screenshot 2024-06-07 193933

Screenshot 2024-06-07 195335

Screenshot 2024-06-07 195915

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Todo:
- [x] Don't instantiate font in frame update

**Changelog**

Fixes #28677

### CODE REVIEW

**Comment by IProduceWidgets** [comment]
Any chance you can get keybind assignments to show automatically?

File: Content.Client/MapText/MapTextOverlay.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+            if (mapPos.MapId != args.MapId)
+                continue;
+
+            if (!args.WorldBounds.Contains(mapPos.Position))
+                continue;
+
+            var fontPrototype = _prototypeManager.Index<FontPrototype>(mapText.FontId);
+            var fontResource = _resourceCache.GetResource<FontResource>(fontPrototype.Path);
+            var font = new VectorFont(fontResource, mapText.FontSize);
```
Comment (ElectroJr) [inline Content.Client/MapText/MapTextOverlay.cs:80]: Maybe cache this on the component if you can? Though that might require splitting the comp over server & client

**Comment by ElectroJr** [comment]
> Any chance you can get keybind assignments to show automatically?

If this gets extended to render arbitrary controls, you could have it draw guides entries or rich text., and IIRC guides support displaying keybinds. Though not sure how much effort that will be, and it might not get done in time for getting a quick tutorial area in.

**Comment by juliangiebel** [comment]
> > Any chance you can get keybind assignments to show automatically?
> 
> If this gets extended to render arbitrary controls, you could have it draw guides entries or rich text. and IIRC guides support displaying keybinds. Though not sure how much effort that will be, and it might not get done in time for getting a quick tutorial area in.

I tried and failed doing that. It's probably possible but not in time for the pc gamer tide 😔

**Review by ElectroJr — APPROVED** [review]

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**